python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -98,6 +98,8 @@ def readMetadata(self):
         metadata : `~lsst.daf.base.PropertyList`
             Header metadata.
         """
+        # Do not use ExposureFitsReader.readMetadata because that strips
+        # out lots of headers and there is no way to recover them
```

Comment:
Does this mean we don't do that stripping anymore when someone asks for `.metadata`, rather than something else?

Maybe that's for the best, given the trouble stripping has caused, but I'd be surprised if it _didn't_ have knock-on effects.

Reply:
Maybe we need a fullMetadata "read-only" component.  The raw exposure has to use the full read so at minimum we'd have to move this code there instead. It's possibly confusing that you get a difference answer getting the .metadata component rather than getting the exposure and asking for the metadata 😄

Reply:
That's not a bad solution.  Thinking about it a bit further, there probably wouldn't be knock-on effects, because I think this brings the direct-metadata access back to what Gen2 does (or at least what it did until very recently - I don't recall if we did eventually switch it to `ExposureFitsReader` or not).  But the extra stripping was very much intentional - it was intended to force people to break a bad habit of doing things like getting the bbox manually by reading the metadata themselves, and then having their code totally broken when we turn on compression.

Reply:
That explains the presence of the stripMetadata method then which strips out bbox and wcs headers. If I return the stripped form by default and the full form with a read-only component I can get rid of the stripping code.